core/node/node_framework/src/implementations/layers/eth_watch.rs

```diff
@@ -69,6 +75,19 @@ impl WiringLayer for EthWatchLayer {
     async fn wire(self, input: Self::Input) -> Result<Self::Output, WiringError> {
         let main_pool = input.master_pool.get().await?;
         let client = input.eth_client.0;
+
+        let (base_token_name, base_token_symbol) =
+            if let Some(base_token_addr) = self.contracts_config.base_token_addr {
+                let (name, symbol) = get_token_metadata(&client, base_token_addr)
+                    .await
+                    .map_err(WiringError::Internal)?;
+                (Some(name), Some(symbol))
+            } else {
+                (None, None)
+            };
```

Comment:
It doesn't belong here. I suggest adding some method eth_watch crate that would return `ZkChainSpecificUpgradeData` by client + other stuff